server/src/main/java/org/opensearch/index/query/TermsQueryBuilder.java

```diff
@@ -251,6 +258,9 @@ public TermsQueryBuilder(StreamInput in) throws IOException {
         termsLookup = in.readOptionalWriteable(TermsLookup::new);
         values = (List<?>) in.readGenericValue();
         this.supplier = null;
+        if (in.getVersion().after(Version.V_3_0_0)) {
```

Comment:
We are targeting 3.0.0 here:

```suggestion
        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Reply:
This is a not a rule, but adding serialization of the new fields at the end greatly reduce the risk of messing up the deserialization (unconsumed content vs possibly weird values)

Reply:
got it!